src/Umbraco.Core/Persistence/Repositories/ContentTypeRepository.cs

```diff
@@ -171,24 +171,37 @@ protected override void PersistNewItem(IContentType entity)
 
             ((ContentType)entity).AddingEntity();
 
-            var factory = new ContentTypeFactory(NodeObjectTypeId);
-            var dto = factory.BuildDto(entity);
+            PersistNewBaseContentType(entity);
+            PersisteTemplates(entity, false);
 
-            PersistNewBaseContentType(dto.ContentTypeDto, entity);
-            //Inserts data into the cmsDocumentType table if a template exists
-            if (dto.TemplateNodeId > 0)
+            entity.ResetDirtyProperties();
+        }
+
+        protected void PersisteTemplates(IContentType entity, bool clearAll)
+        {
+            // remove and insert, if required
+            Database.Delete<ContentTypeTemplateDto>("WHERE contentTypeNodeId = @Id", new { Id = entity.Id });
+
+            // we could do it all in foreach if we assume that the default template is an allowed template??

```

Comment:
Might actually be working like that in the new doctype editor interface. In the old one it was possible to make one default and not allow it.. o_0 - this is OK for now.